src/Graph.cpp

```
@@ -0,0 +1,93 @@
+#include "Graph.hpp"
+
+#include <boost/graph/adjacency_list.hpp>
+#include <unordered_map>
+#include <queue>
+#include <vector>
+#include <string>
+#include <tuple>
+
+namespace booty {
+
+using BoostGraph = boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS>;
+using Vertex = boost::graph_traits<BoostGraph>::vertex_descriptor;
+
+class Graph::GraphImpl {
+public:
+    BoostGraph graph;
+    std::unordered_map<std::string, Vertex> vertexMap;
+    std::unordered_map<Vertex, std::string> labelMap;
+};
+
+Graph::Graph() : impl(new GraphImpl) {}
+
+Graph::~Graph() { delete impl; }
+
+void Graph::add_node(const std::string& label) {
+    if (impl->vertexMap.find(label) == impl->vertexMap.end()) {
```

Currently, if the node already exists, this function silently returns without indicating whether the node was added. It might be helpful to signal whether the node was newly added or already existed.